s3store/s3store.go

```diff
@@ -139,12 +141,17 @@ type S3Store struct {
 	// MaxObjectSize is the maximum size an S3 Object can have according to S3
 	// API specifications. See link above.
 	MaxObjectSize int64
+	// IncompleteUploadTag is a tag that is applied to incomplete parts when they
+	// are stored in S3. This mechanism can be used to create an expiration policy
+	// to remove incomplete objects after a period of time.
+	IncompletePartTag string
```

Comment:
For our use case, we want to keep completed multipart uploads in the bucket but let everything else expire.

The bucket is configured with a policy to remove incomplete multipart uploads after a certain number of hours. However, that leaves the problem of cleaning up these `.part` objects, which are regular S3 objects and thus not attached to a multipart upload. If we support tagging such objects, then we can create a bucket policy that automatically cleans up any object with this tag.

Reply:
> The bucket is configured with a policy to remove incomplete multipart uploads after a certain number of hours.

Makes sense, but how are you dealing with the `.info` objects for the expired incomplete multipart uploads? Do you have another policy to automatically remove them?

Reply:
Ah, sorry for my confusion. We have a separate mechanism for cleaning up those orphaned `.info` objects, but it would be simpler and more reliable to let tusd apply tags to them. Then, a simple expiration policy (possibly the same one) could be used. If you're okay with that approach, I'll apply the tag to `.info` and `.part` objects.

Reply:
Ok, that makes sense. Handling expiration using tags sounds like a good idea but I think this should be part of separate PR which handles upload expiration in a more complete approach. Would you be acceptable with moving this into another PR where we can discuss those tags in more detail?

Reply:
That sounds good to me. I'll rework this branch so that it doesn't include the tagging changes, and then I'll open a separate PR for them.

Reply:
Perfect, thank you very much! I will be on a trip next week, so I might not be able to have a look at any changes but I will catch up with everything the following week.

Reply:
Enjoy your trip :)